android/src/main/java/com/qubit/reactnative/sdk/QubitSDKModule.java

```diff
@@ -139,18 +147,85 @@ public void getExperiences(ReadableArray experienceIds,
 
   @ReactMethod
   public void experienceShown(String callback) {
-    CallbackConnector callbackConnector = new CallbackConnectorImpl(callback, QubitSDK.getDeviceId());
+    ExperienceCallbackConnector callbackConnector = new ExperienceCallbackConnectorImpl(callback, QubitSDK.getDeviceId());
     callbackConnector.shown();
   }
 
+  @ReactMethod
+  public void getPlacement(
+      String placementId,
+      String mode,
+      String attributes,
+      String campaignId,
+      String experienceId,
+      Promise placementPromise
+  ) {
+    QubitSDK.getPlacement(
+        placementId,
+        matchMode(mode),
+        getAttributesJson(attributes),
+        new PlacementPreviewOptions(campaignId, experienceId),
+        placement -> {
+          JsonObject placementJson = placement.getContent();
+          placementJson.addProperty("impressionUrl", placement.getImpressionUrl());
+          placementJson.addProperty("clickthroughUrl", placement.getClickthroughUrl());
+          placementPromise.resolve(WritableMapConverter.convertJsonToMap(placementJson));
+          return null;
+        },
+        throwable -> {
+          placementPromise.reject(throwable);
+          return null;
+        }
+    );
+  }
+
+  @ReactMethod
+  public void placementImpression(String callbackUrl) {
+    PlacementCallbackConnector callbackConnector = new PlacementCallbackConnectorImpl(
```

Comment:
I don't like publishing mysterious getCallbackRequestTracker in public API. It's part of internals.
Two options:
* Leave it as it is, but document it better, that it is not for use.
* Publish sendPlacementCallback(String url) instead. (I would prefer this option)

Reply:
This comment should be in qubit-sdk-android.